src/main/java/com/academia/andruhovich/library/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+package com.academia.andruhovich.library.service.impl;
+
+import com.academia.andruhovich.library.dto.BookDto;
+import com.academia.andruhovich.library.dto.OrderContentWrapper;
+import com.academia.andruhovich.library.dto.OrderResponseDto;
+import com.academia.andruhovich.library.dto.OrderRequestDto;
+import com.academia.andruhovich.library.exception.ResourceNotFoundException;
+import com.academia.andruhovich.library.exception.UpdateDeniedException;
+import com.academia.andruhovich.library.mapper.BookMapper;
+import com.academia.andruhovich.library.mapper.OrderMapper;
+import com.academia.andruhovich.library.model.Book;
+import com.academia.andruhovich.library.model.Order;
+import com.academia.andruhovich.library.model.User;
+import com.academia.andruhovich.library.repository.BookRepository;
+import com.academia.andruhovich.library.repository.OrderRepository;
+import com.academia.andruhovich.library.service.BookService;
+import com.academia.andruhovich.library.service.OrderService;
+import com.academia.andruhovich.library.service.UserService;
+import com.academia.andruhovich.library.util.DateHelper;
+import com.academia.andruhovich.library.util.JsonConverter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.academia.andruhovich.library.exception.ErrorMessages.BOOK_NOT_FOUND;
+import static com.academia.andruhovich.library.exception.ErrorMessages.ORDER_NOT_FOUND;
+import static com.academia.andruhovich.library.exception.ErrorMessages.UPDATE_DENIED;
+import static com.academia.andruhovich.library.model.OrderStatus.DRAFT;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final JsonConverter jsonConverter;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final UserService userService;
+    private final BookService bookService;
+
+    @Transactional
+    @Override
+    public Set<OrderResponseDto> getAll() {
+        User user = userService.getCurrent();
+        Set<Order> orders = orderRepository.getAllByUserId(user.getId());
+        return orders.stream()
+                .map(this::isNeedRecalculate)
+                .map(orderMapper::modelToDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Transactional
+    @Override
+    public OrderResponseDto getOrder(Long id) {
+        User user = userService.getCurrent();
+        Order order = orderRepository.getByIdAndUserId(id, user.getId()).orElseThrow(() ->
+                new ResourceNotFoundException(String.format(ORDER_NOT_FOUND, id)));
+        if (DRAFT.equals(order.getStatus())) {
```

Comment:
copy past with ?

private Order isNeedRecalculate(Order order) {
        if (DRAFT.equals(order.getStatus())) {
            return recalculatePriceAndHistory(order);
        }
        return order;
    }

Reply:
Refactored it